### PR TITLE
Requirements

### LINKED ISSUE #43: Update install and requirements
Check dependencies and their versions

### PR DESCRIPTION
**Pull request by a-arefina**
fixes #43 

* PyWavelets removed from code and requirements
* setup.py updated to match requirements and also includes batchflow (there might be a better way to indicate the bf revision)
* test data moved inside package to allow running tests from an installed package

### CODE REVIEW

**Comment by a-arefina** [comment]
not quite sure if both install_with_pip and install_with_pipenv are really needed

File: requirements.txt [hunk 1]
```diff
@@ -1 +1,9 @@
-.
\ No newline at end of file
+matplotlib>=3.3.1
```
Comment (roman-kh) [inline requirements.txt:1]: AFAIR, you can put `.` in req.txt, in order not to duplicate `install_requires`

File: setup.py [hunk 2]
```diff
@@ -18,19 +18,23 @@
...
+        'scikit-learn>=0.23.2',
+        'scipy>=1.5.2',
+        'segyio>=1.9.5',
         'tdigest>=0.5.2.2',
-        'tqdm==4.30.0',
-        'segyio==1.8.3',
-        'scikit-image>=0.13.1',
-        'numba>=0.35.0'
+        'tqdm>=4.56.0',
+        'batchflow @ git+https://github.com/analysiscenter/batchflow.git@9823f369#egg=batchflow',
```
Comment (roman-kh) [inline setup.py:33]: It seems pretty legit to me

File: setup.py [hunk 3]
```diff
@@ -18,19 +18,23 @@
...
+
+        # 'dill>=0.2.7.1',
+        # 'scikit-image>=0.13.1',
+        # 'numba>=0.35.0'
     ],
     extras_require={
         'tensorflow': ['tensorflow>=1.12'],
```
Comment (roman-kh) [inline setup.py:40]: Do we need `tf`? `2.2` > `1.12`

**Comment by roman-kh** [comment]
> not quite sure if both install_with_pip and install_with_pipenv are really needed

Install order might differ, so it's better to reveal problems in advance.

File: .github/workflows/test-install.yml [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+# -----------------------------------------
+#    Use a module from local source
+# -----------------------------------------
+  use_as_local_module:
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest] #, macos-latest, windows-latest]
```
Comment (nikita-klsh) [inline .github/workflows/test-install.yml:18]: Should we consider the loop over all OS's since we have classifier `Operating System :: OS Independent`?

Comment (roman-kh) [reply .github/workflows/test-install.yml:18]: Even though pytorch supports ROCm, the latter does not support MacOS yet. So we can definitely exclude MacOS for now.

**Review by dpodvyaznikov — APPROVED** [review]

**Comment by dpodvyaznikov** [comment]
If there are no objections, I'd merge this by the end of the day.

**Pull request merged**